### DISCUSSION TITLE
Is there a tracking API available?

### DISCUSSION THREAD

**Discussion by uhlhosting** [postalserver/postal#2583] [Development discussions]
Can we use somehow the tracking feature to get a status back with the Postal php library when a email is opened/read and has tracking enabled?

**Comment by willpower232** [comment]
If you have the tracking working then all you need to do is start handling webhooks and Postal will be able to send you a message when things happen https://docs.postalserver.io/developer/webhooks

**Reply by uhlhosting** [reply]
Thanks for the tip, why did I not think of that, I wish to have this as a extra feature to add in my WordPress plugin. Btw have you seen I made a release, including Postman API library, well they are still WIP, anyhow more details here: https://gitlab.uhlhost.net/cosmin/postal-mail feel free to share alike. I was a bit shocked there were no plugins to allow sending mails with Postal in WP.

**Reply by willpower232** [reply]
In order to track emails you would first need to store a copy of the emails that have been sent but I'm not sure if you're "allowed" to create an email specific table in your wordpress database or if they would rather you store everything as a type of "post".

Once you are storing the messages then you would need to present a way of receiving webhooks in your code through wordpress, again, not sure how you would do that.

Unfortunately I am not a wordpress developer so I don't really know the answers to the above but if you can figure out those two complicated parts, then the rest should be relatively straightforward.

I can imagine most people would use SMTP rather than the API as SMTP tends to be integrated everywhere already so just needs some details but that does cut you off from using webhooks and tracking properly.

**Reply by uhlhosting** [reply]
I already store this data that is then showed to users like last 5 sent emails. So the data that you see bellow I already store it in SQL own plugins db.

This is only what I could get back from the server.

```
Email: cosmin@uhlhost.net, Message ID: 2608, Sent at: 2023-08-19 14:01:20
Email: cosmin@uhlhost.net, Message ID: 2598, Sent at: 2023-08-18 14:01:16
Email: cosmin@uhlhost.net, Message ID: 2594, Sent at: 2023-08-17 14:01:17
Email: cosmin@uhlhost.net, Message ID: 2588, Sent at: 2023-08-16 13:57:03
Email: cosmin@uhlhost.net, Message ID: 2475, Sent at: 2023-08-12 02:32:24
```

**Reply by uhlhosting** [reply]
The issue is with SMTP i never managed to get it working with PHPmailer settings from WP, since wp uses phpmailer to send mails... So I thought API is more robust, and for our needs mostly transactional mails works wonders.

If there is someone who made Postal work with PHPmailer settings and willing to share details I am more than happy to give it a try, otherwise I am forced to use the API. And I guess the PHP library handles only API not also SMTP from what I have seen.

**Reply by willpower232** [reply]
if you have the message IDs then you just need a way of receiving webhooks and you can show the tracking information to users.

The libraries don't tend to add SMTP functionality because that is almost always provided in a different way and there is nothing Postal specific about sending emails by SMTP. Using the API means you can easily store the message ID which is crucial for using webhooks.

**Reply by uhlhosting** [reply]
@willpower232 Thanks for all the tips and info. Will surely try some methods. I will also add the smtp method using php mailer. And smtp ip, even since that might work just fine too.
